src/common/page_ir_macro.h

```diff
@@ -11,4 +16,20 @@
     return reinterpret_cast<type *>(page->ptr)[idx]; \
   } \
 
-#endif // PAGE_IR_MACRO_H_
\ No newline at end of file
+#define READ_RLE_VAL(suffix, type) \
+  extern "C" type read_##suffix##_rle(RLEChunk* chunk, size_t idx) { \
+    int16_t n = chunk->run_num; \
+    int16_t i; \
+    size_t r = 0; \
+    for (i = 0; i < n; i++) { \
+      Run run = chunk->runs[i]; \
+      if (r <= idx && idx < r + *(run.length)) { \
+        return *(reinterpret_cast<type *>(run.value)); \
+      } \
+      r += *(run.length); \
+    } \
+    /* TODO: error handling */ \
+    return -1; \
```

Comment:
If we can ensure that the case is impossible, no return may be good. This may be only good case for 'no return'. Otherwise, how about ``std::abort()``?